system-tests/e2e-testsupport/src/test/java/com/hotels/styx/servers/WiremockStyxRequestAdapterTest.java

```diff
@@ -0,0 +1,179 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.servers;
+
+import com.github.tomakehurst.wiremock.http.ContentTypeHeader;
+import com.github.tomakehurst.wiremock.http.HttpHeader;
+import com.github.tomakehurst.wiremock.http.HttpHeaders;
+import com.github.tomakehurst.wiremock.http.QueryParameter;
+import com.github.tomakehurst.wiremock.http.RequestMethod;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+
+import static com.hotels.styx.api.HttpHeaderNames.CONNECTION;
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_LENGTH;
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_TYPE;
+import static com.hotels.styx.api.HttpRequest.Builder.post;
+import static io.netty.handler.codec.http.HttpHeaderNames.HOST;
+import static io.netty.handler.codec.http.HttpHeaderNames.USER_AGENT;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.testng.Assert.assertEquals;
+
+public class WiremockStyxRequestAdapterTest {
+    private WiremockStyxRequestAdapter adapter;
+    private String content;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        content = "{\n" +
+        "    \"request\" : {\n" +
+        "        \"urlPattern\" : \"/.*\",\n" +
+        "        \"method\" : \"GET\"\n" +
+        "    },\n" +
+        "    \"response\" : {\n" +
+        "        \"status\" : 200,\n" +
+        "        \"body\" : \"Hello, World!\",\n" +
+        "        \"headers\" : {\n" +
+        "            \"Stub-Origin-Info\" : \"App TLS v1.1\"\n" +
+        "        }\n" +
+        "    }\n" +
+        "}";
+
+        adapter = new WiremockStyxRequestAdapter(
+                post("/__admin/mappings/new?msg=6198.1")
+                .header(CONTENT_LENGTH, "208")
+                .header(CONTENT_TYPE, "application/json; charset=UTF-8")
+                .header(HOST, "localhost")
+                .header(CONNECTION, "Keep-Alive")
+                .header(USER_AGENT, "Apache-HttpClient/4.3.5 (java 1.5)")
+                .body(content)
+                .build(), content);
+    }
+
+    @Test
+    public void adaptsContainsHeader() throws Exception {
+        assertThat(adapter.containsHeader("Foo-Bar"), is(false));
+        assertThat(adapter.containsHeader("Host"), is(true));
+        assertThat(adapter.containsHeader("host"), is(true));
+    }
+
+    @Test(enabled = false)
```

Comment:
Disabled tests should not be added to the codebase.

Reply:
I have fixed the tests.

Reply:
Except for the one directly above these comments.